### ISSUE TITLE
Support passing the path to the ESLint module

### ISSUE THREAD

**Issue by insin** [MoOx/eslint-loader#153]
Allowing the path to the ESLint module to be passed would support use of this loader with [reusable modules which also manage ESLint dependencies for you](https://github.com/insin/eslint-config-jonnybuchanan#readme), e.g.:

```js
var path = require('path')
var resolve = require('resolve')
var webpackConfig = {
  module: {
    rules: [
      {
        test: /\.js$/,
        enforce: 'pre',
        loader: 'eslint-loader',
        options: {
          configFile: require.resolve('reusable-eslint-module/.eslintrc'),
          eslintPath: resolve.sync('eslint', {basedir: path.join(__dirname, 'reusable-eslint-module')})
        },
        exclude: /node_modules/
      }
    ]
  }
}
```

Implementation-wise, ESLint could be imported only when needed (and similarly in the `lint()` function):

```diff
   // Create the engine only once per config
   var configHash = objectHash(config)
   if (!engines[configHash]) {
+    var eslint = require(config.eslintPath || 'eslint')
     engines[configHash] = new eslint.CLIEngine(config)
   }
```

**Comment by MoOx** [comment]
Why not :)

**Comment by trungdq88** [comment]
I have submitted a PR here https://github.com/MoOx/eslint-loader/pull/181

@MoOx do you mind have a look?

**Comment by MoOx** [comment]
Closed by #181
